src/tools/rbd_mirror/image_map/types.h

```diff
@@ -0,0 +1,161 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RBD_MIRROR_IMAGE_MAP_TYPES_H
+#define CEPH_RBD_MIRROR_IMAGE_MAP_TYPES_H
+
+#include <set>
+#include <map>
+#include <string>
+#include <memory>
+
+namespace rbd {
+namespace mirror {
+namespace image_map {
+
+enum MapShuffleType {
+  SHUFFLE_TYPE_INSTANCES_ADDED = 0,   // rebalance due to addition of instances
+  SHUFFLE_TYPE_INSTANCES_REMOVED,     // rebalance due to removal of instances
+  SHUFFLE_TYPE_IMAGES_ADDED_REMOVED,  // rebalance due to adding/removing images
+};
+
+struct MappedImage {
+  std::string id;
```

Comment:
The `id` here is not for tracking. This structure is used by `PoolReplayer::handle_update()` and `id` gets assigned to whatever we get in image_id in `PoolWatcher` (although as you mentioned that passing `id` to start image replayers is not really required as we already pass the `global_id`). This can be taken away if required.